util/types/convert.go

```diff
@@ -137,79 +146,152 @@ func isCastType(tp byte) bool {
 	return false
 }
 
-// StrToInt converts a string to an integer in best effort.
-// TODO: handle overflow and add unittest.
+// StrToInt converts a string to an integer at the best-effort.
 func StrToInt(sc *variable.StatementContext, str string) (int64, error) {
 	str = strings.TrimSpace(str)
-	if len(str) == 0 {
-		return 0, nil
-	}
-	negative := false
-	i := 0
-	if str[i] == '-' {
-		negative = true
-		i++
-	} else if str[i] == '+' {
-		i++
-	}
-	r := int64(0)
-	for ; i < len(str); i++ {
-		if !unicode.IsDigit(rune(str[i])) {
-			break
+	validPrefix, err := getValidIntPrefix(sc, str)
+	iVal, err1 := strconv.ParseInt(validPrefix, 10, 64)
+	if err1 != nil {
+		return iVal, errors.Trace(ErrOverflow)
+	}
+	return iVal, errors.Trace(err)
+}
+
+// StrToUint converts a string to an unsigned interger at the best-effortt.
+func StrToUint(sc *variable.StatementContext, str string) (uint64, error) {
+	str = strings.TrimSpace(str)
+	validPrefix, err := getValidIntPrefix(sc, str)
+	if validPrefix[0] == '+' {
+		validPrefix = validPrefix[1:]
+	}
+	uVal, err1 := strconv.ParseUint(validPrefix, 10, 64)
+	if err1 != nil {
+		return uVal, errors.Trace(ErrOverflow)
+	}
+	return uVal, err
+}
+
+// getValidIntPrefix gets prefix of the string which can be successfully parsed as int.
+func getValidIntPrefix(sc *variable.StatementContext, str string) (string, error) {
+	floatPrefix, err := getValidFloatPrefix(sc, str)
```

Comment:
Because valid floatPrefix like `1.1e10` can not be parsed by `strconv.ParseInt`.